### ISSUE TITLE
Changing theme in one tab make widgets look broken on another one.

### ISSUE THREAD

**Issue by alexballas** [fyne-io/fyne#2996]
### Describe the bug:
When I change the theme from one App tab, the widgets on the other tabs look broken

### To Reproduce:
- Go to tab 2, select either the dark or light variant and observe the widgets on tab 1 (check code example below)

### Screenshots:
https://i.imgur.com/t0PISE7.png
https://i.imgur.com/t9UhgVx.png
https://i.imgur.com/MxtxN1Q.png

### Example code:
```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/theme"
	"fyne.io/fyne/v2/widget"
)

func main() {
	myApp := app.New()
	myWindow := myApp.NewWindow("Test")

	content := widget.NewButton("Just a button", func() {})

	light := widget.NewButton("Light Theme", func() {
		fyne.CurrentApp().Settings().SetTheme(theme.LightTheme())
	})
	dark := widget.NewButton("Dark Theme", func() {
		fyne.CurrentApp().Settings().SetTheme(theme.DarkTheme())
	})

	tabs := container.NewAppTabs(
		container.NewTabItem("Tab 1", container.NewHBox(makeFormTab(myWindow), content)),
		container.NewTabItem("Tab 2 Set Theme", container.NewVBox(light, dark)),
	)
	myWindow.SetContent(tabs)
	myWindow.ShowAndRun()

}

func makeFormTab(_ fyne.Window) fyne.CanvasObject {
	name := widget.NewEntry()
	name.SetPlaceHolder("John Smith")

	email := widget.NewEntry()
	email.SetPlaceHolder("test@example.com")

	password := widget.NewPasswordEntry()
	password.SetPlaceHolder("Password")

	disabled := widget.NewRadioGroup([]string{"Option 1", "Option 2"}, func(string) {})
	disabled.Horizontal = true
	disabled.Disable()
	largeText := widget.NewMultiLineEntry()

	form := &widget.Form{
		Items: []*widget.FormItem{
			{Text: "Name", Widget: name, HintText: "Your full name"},
			{Text: "Email", Widget: email, HintText: "A valid email address"},
		},
	}
	form.Append("Password", password)
	form.Append("Disabled", disabled)
	form.Append("Message", largeText)
	return form
}

```
-->

### Device (please complete the following information):
 - **OS:** Linux Ubuntu 18.04
 - **Go version:** v1.18.2
 - **Fyne version:** v2.1.4

**Comment by Jacalz** [comment]
Can you please try the develop branch? I think this might be a duplicate of https://github.com/fyne-io/fyne/issues/2873.

**Comment by alexballas** [comment]
Thank you @Jacalz ,
I switched to develop but the issue still persists.

**Comment by alexballas** [comment]
Explicitly cycling through the tab items and refreshing them seems to do the trick

```go
            for _, tab := range s.tabs.Items {
                tab.Content.Refresh()
            }
```

**Comment by alexballas** [comment]
this approach / workaround seems more robust
```go
	s.tabs.OnSelected = func(t *container.TabItem) {
		t.Content.Refresh()
	}
```

**Comment by BabySid** [comment]
> this approach / workaround seems more robust
> 
> ```go
> 	s.tabs.OnSelected = func(t *container.TabItem) {
> 		t.Content.Refresh()
> 	}
> ```

I also encountered this problem. I have use both of the above methods to solve the problem. But the most direct problem is that if the selection event is not triggered, the UI looks ugly..

**Comment by alexballas** [comment]
What I found working better is this.
```go
	s.tabs.OnSelected = func(t *container.TabItem) {
		fyne.CurrentApp().Settings().SetTheme.....
	}
```
So instead of calling refresh, actually reset the theme.

**Comment by andydotxyz** [comment]
This was resolved in v2.5.0, sorry that I missed this issue update!
